modules/images/image-loading-optimization/detection.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Detection for image loading optimization.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
```

Comment:
There are a lot of functions in this file, many of which seem low-level and would be good to be private.

I think at a minimum, we should mark those functions private that we don't want external code to use. Alternatively, we could refactor into a simple class to actually use private methods, but if so, that should probably be done in a follow up PR to not have to focus on refactoring _and_ the actual logic at the same time.

Reply:
Do you mean the file `storage/data.php`?

And yeah, none of these functions should need to be accessed by developers.

Reply:
I've marked all functions as being private in 19ce752.